### DISCUSSION TITLE
Nuxt useFetch returns JSON object on dev but JSON string on prod

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#28540] [Questions] [answered]
Currently running into an issue with Nuxt & Netlify. Dev works perfectly using the following code as the response is automatically converted to a JSON object. However on Prod, the `response._data.products` returns as a JSON string. JSON.parse will get it to work on Prod with a bit of hacking to the pinia store, but it then causes it to fail on Dev.

```js
const { data, error } = await useFetch(`${useRuntimeConfig().public.MAIN_URL}/.netlify/functions/getProducts`, {
  onResponse({ request, response, options }) {
    // Process the response data
    productStore.products = response._data.products;
  },
  onResponseError({ request, response, options }) {
    // Handle the response errors
    console.error(response);
  },
  key: Date.now().toString(),
});
```

![Screenshot 2024-08-14 at 10 14 15](https://github.com/user-attachments/assets/718b2f78-e3e4-4fb9-95ea-3b79eb02c9f8)

Production Response

![Screenshot 2024-08-14 at 10 15 21](https://github.com/user-attachments/assets/2bd56798-711f-4a3c-b5c7-1fc53bfda73d)

Development Response

**Comment by TheAlexLichter** [comment]
Probably the production netlify FN doesn't send the application/json Content-Type header?
You can work around that by forcing the response type to convert via `responseType: 'json'` in the options object of use fetch.

A few more notes:
* The key shouldn't be randomly generated 🙈
* I'd suggest to use Nitro (/server folder) if you need an API/a backend anyway

**Reply by ghost** [reply, marked as answer]
Hi there, 

So it turns out useFetch auto stringifies the response, as well as my API so it was clashing heads. I had to move from useFetch to $fetch.raw to fix it.
